tokens/breakpoints.yml

```diff
@@ -0,0 +1,61 @@
+breakpoint:
+  $type: dimension
+  $extensions:
+    com.redhat.ux:
+      use: breakpoint
+      heading: Breakpoints
+      description: |
+        When layouts reduce in size, visual elements may also reduce in size, hide, or otherwise
```

Comment:
**&lt;soap box warning/&gt;**

I feel like this description says we design with a **desktop first** approach.   If this is true then we need to make that distinction. 
> "When layouts reduce in size..."  

Rather than:  "As layout increase in size..." as al la a **mobile-first** methodology.   

It has always been my preference to take a mobile-first approach to layouts being additive in their CSS rules as the layout gets bigger and potentially more complex.  Because of this preference, I usually do not find many use cases for `max-width` rules.  

With that understanding, my translation of these token values would be represented with this table:

| Device           |  Value |
| ------------- | ------------- |
| Mobile portrait | < 576px | 
| Mobile landscape | ≥ 576px |
| Tablet portrait | ≥ 768px |
| Tablet ladscape | ≥ 992px |
| Desktop small | ≥ 1200px |
| Desktop medium + | ≥ 1440px |
| Desktop large + | ≥ 1680px |

Reply:
the description is mostly me riffing on copypasta from ux-dot. commit with wild abandon, as you will.

Reply:
as for the token names, as per my usual, i'm ok with either. maybe slightly edging towards Sprigg's names, but OTOH developers typically need to make a mobile, tablet, and desktop version, not a small, med, and large version. :shrug:

Reply:
I would also submit that most major CSS frameworks use `small`, `medium`, and `large` these days for breakpoints.  Not that we have to be like everyone else but there is definitely a precedent.

[Bootstrap](https://getbootstrap.com/docs/5.0/layout/breakpoints/), [Tailwind](https://tailwindcss.com/docs/screens), [Foundation](https://get.foundation/sites/docs/media-queries.html), [UIKit](https://getuikit.com/v2/docs/core.html), [Carbon Design System](https://get.foundation/sites/docs/media-queries.html) to name a few

Reply:
@zeroedin Love this! Your names and values look great to me. Do you care to update the preview page?

Reply:
So, this is a great soapbox!

I agree it should be reworded to be "mobile-first" or at least agnostic:

> As layouts increase in size...

...or...

> As layouts change in size...

As for the device breakpoint descriptors, I can see both sides to this. On one hand, `mobile`, `desktop`, etc. are very commonly used and will mostly likely feel more intuitive to developers due to whatever potential bad habits we've developed over time with this nomenclature (thanks to libraries like Bootstrap). On the other hand, they're poor descriptors because screen sizes are not solely reliant on the type device or how someone commonly uses it. For instance, `what does it mean to be "mobile"?` or how do you differentiate large mobile screen, or tablets or TVs or small laptops or other touch screen devices that don't fit `desktop` or `mobile` or whatever.

With that, I'm thinking that the right approach might be to define using size (e.g. `small`, `medium`, `large`, etc.). And, if we so desire, then we could create aliases that feel familiar (e.g. `mobile`, `tablet`, etc.)—with an SLA that these could be deprecated or **are** deprecated out of the gate (so people will hopefully change their mindset).

Reply:
I did a few things:

- renamed the tokens to `2xs`, `xs`, `sm`, etc (in keeping with other sizes)
- refactored the sources to use dtcg syntax (`$value` instead of `value`, etc)
- moved some tokens around for maintenance reasons
- added a JSON file to the page downloads
I know most of that is out of scope. I have no excuses :see_no_evil:

Reply:
gonna hold release until @zeroedin can make his changes